error_context.go

```diff
@@ -14,8 +14,9 @@ func DetailedError(err error) string {
 		msg.WriteString(fmt.Sprintf("(%s:%d) ", f, l))
 	}
 	msg.WriteString(err.Error())
-	for werr := Unwrap(err); werr != nil; werr = Unwrap(werr) {
-		msg.WriteString(fmt.Sprintf("\nCaused by: %s\n", err.Error()))
+	msg.WriteString("\n")
+	if werr := Unwrap(err); werr != nil {
+		msg.WriteString(fmt.Sprintf("Caused by: %s", DetailedError(werr)))
```

Comment:
@manini-flare I think the main bug you're trying to solve still exists with this, `DetailedError` has the same bugged logic

https://github.com/ojizero/errors/blob/5ac5a7bc5af050f27af9849441f630b92363b80d/error_context.go#L18

Reply:
the main error is fixed in the first commit to make it clear it was using the **err** instead of the **werr** in the loop and that cause the same error msg printed for the wrapped errors

Reply:
the second commit is just for formatting instead of duplicating the logic to extract the msg i used the samefunction call

Reply:
oh just noticed it was doing a recursion in place of the original loop my bad 🤦🏻‍♂️